sertop/sercomp.ml

```diff
@@ -187,7 +192,8 @@ let driver input mode debug printer async async_workers quick coq_path ml_path l
   (* main loop *)
   let in_chan = open_in in_file in
   let doc = input_doc ~input ~in_file ~in_chan ~process ~doc ~sid in
-  close_document ~mode ~doc ~in_file
+  let () = close_document ~pp ~mode ~doc ~in_file in
```

Comment:
Not sure I understand this idiom. Does this mean every `unit`-returning function should end in `()`?

Reply:
Very good question, actually this is a kind of corner case I'd like to guard against [warnings have been improved in OCaml 4.08 tho].

If you omit for example the `~pp` argument, in this case OCaml will just infer the type of driver to be a function, and the close document won't be performed; the `let () =` does indeed force the typechecker to check that the function is of type unit.